pkg/planner/core/task.go

```diff
@@ -978,18 +1077,46 @@ func (p *PhysicalTopN) Attach2Task(tasks ...base.Task) base.Task {
 		// If all columns in topN are from index plan, we push it to index plan, otherwise we finish the index plan and
 		// push it to table plan.
 		var pushedDownTopN *PhysicalTopN
+		var newGlobalTopN *PhysicalTopN
 		if !copTask.indexPlanFinished && p.canPushToIndexPlan(copTask.indexPlan, cols) {
-			pushedDownTopN = p.getPushedDownTopN(copTask.indexPlan)
+			pushedDownTopN, newGlobalTopN = p.getPushedDownTopN(copTask.indexPlan, copTask.getStoreType())
```

Comment:
I think it is need. after we add a new column to save the distance column, the result can be re-use in tidb, and It can further avoid unnecessary exchange of vector columns.